framework/src/systems/AuxiliarySystem.C

```diff
@@ -736,12 +736,20 @@ AuxiliarySystem::computeMortarNodalVars(const ExecFlagType type)
       {
         PARALLEL_TRY
         {
-          ComputeMortarNodalAuxBndThread<AuxKernel> mnabt(_fe_problem, mortar_nodal, bnd_id, index);
-          Threads::parallel_reduce(bnd_nodes, mnabt);
-          solution().close();
-          _sys.update();
+          try
+          {
+            ComputeMortarNodalAuxBndThread<AuxKernel> mnabt(
+                _fe_problem, mortar_nodal, bnd_id, index);
+            Threads::parallel_reduce(bnd_nodes, mnabt);
+          }
+          catch (MooseException & e)
+          {
+            _fe_problem.setException(e.what());
+          }
```

Comment:
ComputeMortarNodalAuxBndThread
```suggestion
          catch (MooseException & e)
          {
            _fe_problem.setException(e.what());
          }
          catch (libMesh::LogicError & e)
          {
            _fe_problem.setException("We caught a libMesh::LogicError.");
          }
```

Reply:
Yeah, we can catch the `libMesh::LogicError` in `ComputeMortarNodalAuxBndThread` and do nothing in `FEProblemBase` to avoid parallel communication issues. That seems to work.